hazelcast/src/main/java/com/hazelcast/dataconnection/impl/JdbcDataConnection.java

```diff
@@ -55,45 +59,53 @@
 public class JdbcDataConnection extends DataConnectionBase {
 
     private static final AtomicInteger DATA_SOURCE_COUNTER = new AtomicInteger();
-
     private volatile ConcurrentMemoizingSupplier<HikariDataSource> pooledDataSourceSup;
     private volatile Supplier<Connection> singleUseConnectionSup;
 
     public JdbcDataConnection(DataConnectionConfig config) {
         super(config);
+        validate(config);
         if (config.isShared()) {
             this.pooledDataSourceSup = new ConcurrentMemoizingSupplier<>(this::createHikariDataSource);
         } else {
             this.singleUseConnectionSup = createSingleConnectionSup();
         }
     }
 
+    private void validate(DataConnectionConfig config) {
```

Comment:
If data connection with incorrect params is added through SQL like:
```
sql> CREATE DATA CONNECTION mysql TYPE JDBC OPTIONS('wrongJbcUrl'='whatever');
OK
```

then Exception like following is thrown to server log every second:
```
2023-05-03 08:55:16,144 [ WARN] [heuristic_wu-query-state-checker] [c.h.s.i.s.QueryStateRegistryUpdater]: [192.168.1.115]:5701 [clusterA] [5.3.0-SNAPSHOT] jdbcUrl property is not defined for data connection 'mysql'
com.hazelcast.core.HazelcastException: jdbcUrl property is not defined for data connection 'mysql'
	at com.hazelcast.dataconnection.impl.JdbcDataConnection.validate(JdbcDataConnection.java:78) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.dataconnection.impl.JdbcDataConnection.<init>(JdbcDataConnection.java:67) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at sun.reflect.GeneratedConstructorAccessor14.newInstance(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_311]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_311]
	at com.hazelcast.dataconnection.impl.DataConnectionServiceImpl.createDataConnectionInstance(DataConnectionServiceImpl.java:144) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.dataconnection.impl.DataConnectionServiceImpl.lambda$put$2(DataConnectionServiceImpl.java:103) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1853) ~[?:1.8.0_311]
	at com.hazelcast.dataconnection.impl.DataConnectionServiceImpl.put(DataConnectionServiceImpl.java:83) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.dataconnection.impl.DataConnectionServiceImpl.createOrReplaceSqlDataConnection(DataConnectionServiceImpl.java:109) ~[hazelcast-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.sql.impl.DataConnectionConsistencyChecker.check(DataConnectionConsistencyChecker.java:79) ~[hazelcast-sql-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.sql.impl.state.QueryStateRegistryUpdater$Worker.checkDataConnectionsConsistency(QueryStateRegistryUpdater.java:172) ~[hazelcast-sql-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.sql.impl.state.QueryStateRegistryUpdater$Worker.executeInterruptibly(QueryStateRegistryUpdater.java:148) [hazelcast-sql-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at com.hazelcast.sql.impl.state.QueryStateRegistryUpdater$Worker.run(QueryStateRegistryUpdater.java:132) [hazelcast-sql-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_311]
```

Moreover if this data connection is trying to be used in mapping then it prints incorrect message to user:
```
sql> CREATE MAPPING people DATA CONNECTION "mysql";
Data connection 'mysql' not found
```

Reply:
Maybe it can be fixed in another PR. Otherwise this PR is going to get bigger. @frant-hartm what do you think ?

Reply:
I don't agree that we could fix it in the next PR since this is an issue caused by this PR. It's not another feature or so.